collector/lib/CollectorConfig.h

```diff
@@ -93,6 +108,14 @@ class CollectorConfig {
 
   static std::pair<option::ArgStatus, std::string> CheckConfiguration(const char* config, Json::Value* root);
 
+  void SetRuntimeConfig(sensor::CollectorConfig runtime_config) {
+    runtime_config_ = std::move(runtime_config);
+  }
+
+  std::optional<sensor::CollectorConfig> GetRuntimeConfig() const {
```

Comment:
Lets not copy the object every time it's accessed.
```suggestion
  const std::optional<sensor::CollectorConfig>& GetRuntimeConfig() const {
```